File: Pattern_cpp codes/palindrome.cpp
//palindrome pattern
#include<iostream>
using namespace std;

int main()
{
    int n,i,j ;
    cout<<"enter the  value"<<endl;
    cin>>n;
    for( i=1; i<=n; i++)
    {
        int k=i;
        for( j=1;j<=(n-i);j++)
        {
            cout<<"  ";
        }
        for(; j<=n; j++)
        {
            cout<<k<<" ";
            k--;
        }
        k=1;
        for(; j<(n+i); j++)
        {
            k++;
            cout<<k<<" ";
        }
        for(; j<=(2*n-1); j++)
        {
            cout<<"  ";
        }

        cout<<endl;
    }

    return 0;
}

File: java/palindrome.java
import java.util.*;
class Palindrome {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter a string:");
        String s = sc.nextLine();
        String s1="";

       for (int i = 0; i <= s.length() - 1; i++)
       {
           s1=s.charAt(i)+s1;
        }
        if (s.compareTo(s1)==0) {
            System.out.println("Palindrome");
        } else {
            System.out.println(" Not Palindrome");
        }
    }
}